roles/core/dns_server/templates/named.conf.j2

```diff
@@ -1,13 +1,24 @@
+#jinja2: lstrip_blocks: True
 #### Blue Banquise file ####
 ## {{ansible_managed}}
 
 options {
   listen-on port 53 {
     127.0.0.1;
 {% for network in (networks | select('match',(j2_current_iceberg_network+'-[0-9]+')) | list | unique | sort) %}
-{% if networks[network]['is_in_dns'] is defined and networks[network]['is_in_dns'] == true %}
-    {{networks[network]['services_ip']['dns_ip']}};
-{% endif %}
+  {% if networks[network]['is_in_dns'] is defined and networks[network]['is_in_dns'] == true %}
+    {% set addresses = networks[network]['services_ip']['dns_ip'] | ipaddr %}
+    {% if dns_slaves is defined %}
+      {% set addresses = addresses|union(dns_slaves) %}
+    {% endif %}
+    {% if addresses is string and addresses in ansible_all_ipv4_addresses %}
```

Comment:
Let me explain here in details what makes me not confident about this.

I understand your target here : checking if ip exist on current host, and if yes, add them in the listen part.

However, I would prefer to use the hostvars['network_interfaces'] part. It may appends that for any reason, an interface is down, or an ip is not set on the dns node. Using hostvars['network_interfaces'] would make the named service fail to start as all expected interfaces are not up, and force admin to check what is going on.

If we rely on facts like ansible_all_ipv4_addresses, configuration is dynamically generated, but this also means that configuration deployed could not be always the expected one.
For example an admin deploying dns before configuring all the network, then start service, and configure network after that. dns will not listen on all expected interfaces.

Not sure it is clear. I often prefer a service that fails vs a service that __may__ not provide expected features (like I prefer a nice segmentation fault in C vs a code that work but may not provide the good result).

Reply:
I get your point. Let's wait for additional reviews/comments.

> If we rely on facts like ansible_all_ipv4_addresses, configuration is dynamically generated, but this also means that configuration deployed could not be always the expected one.

This could be the expected configuration. I want to start the dns server on configured interfaces. Some interfaces could be unconfigured on purpose, why should we prevent to start dns?

> For example an admin deploying dns before configuring all the network, then start service, and configure network after that. dns will not listen on all expected interfaces.

The dns server will not listen on all interfaces defined in the inventory until the next run of the playbook.
We might see things differently: I assume the admin knows what he is doing with the system. We do provide example playbooks where network configuration is almost at the beginning of the play.
I don't believe the stack should handle the corner case where the admin configures network services first, then network, then never run the playbook again to ensure the configuration is correct. IMHO, any change to the network configuration should be followed by a run of the playbook on the host to ensure everything is configured as expected.

Reply:
Got the two points

Issue as already been seen and discussed in other roles.

Stack is design to be didactic and 'stupid' for new system administrators, while also trying to be 'smart' and dynamic for advanced users.

Would propose this, not only for this role, but for future:

A variable in inventory, called `local_reference_data`, by default to `facts`, but that could be `inventory`.
Note: facts means all local ways to gather data. May be facts, maybe task command to register, maybe target tool/service dynamic ways to gather data.

All Core and Advanced-Core roles must comply with this variable. Not now, but on long term.

Rule is not mandatory for addons. Addon developer do it the way desired.

Light code overhead

Reply:
OK. For this PR I will revert to inventory instead of facts, as per @oxedions request. Then we can implement the `local_reference_data` in a dedicated PR.

Reply:
Done

Reply:
I opened https://github.com/bluebanquise/bluebanquise/issues/207, so we do not forget.